Param adding additions

milljm opened the pull request:
Works. Now we need a unittest to see how many things this might be breaking...

Refs #26215

Commit:
Param adding additions

Works. Now we need a unittest to see how many things this might be
breaking...

Closes #26273

Commit:
Adding Unittest

Adding Unittest to verify normal tests are no longer skipped if they
are a dependency for a heavy test.

Removed the unittest(s) specifically for non-heavy skippable checks.
As this is a new feature!

Closes #26215

moosebuild commented:
Job [Documentation](https://civet.inl.gov/job/1930285/) on e82cff3 wanted to post the following:

View the site [here](https://mooseframework.inl.gov/docs/PRs/26273/site)

This comment will be updated on new commits.

loganharbour commented on python/TestHarness/JobDAG.py:
```diff
@@ -237,6 +241,32 @@ def _doRaceConditions(self):
+                # TODO: Maybe add a feature to checkRunnableBase, indicating not to cache settings
+                #       so all we have to do is add the missing param
+                prereq_tester.setStatus(prereq_tester.no_status)
+                prereq_tester.addCaveats(f'{param} param added to satisfy dependency')
```
Is also making it a heavy test the only way to get it to run?

milljm replied:
Yeah I like that. What I had was a bit long winded

loganharbour reviewed (commented):
Just a logic question

Commit:
Simple caveat message

Add implicit short and sweet caveat.

Clear the caveats (after adding our implicit caveat). We are
regenerating them. This will remove odd looking combinations like this:

    OK tests/test_harness.not_heavy [NOT HEAVY,Implicit heavy]

Remove the TODO. The feasibility of not caching the testers caveats
and reasons for not running was not going to be worth the effort.

Closes #26215

Commit:
Adjust unittests due to new functionality

Some unittests needed changes to reflect new erro behavior

Commit:
Tests which require mathcing prereqs params

Changes needed due to dependencies now requiring the same
params as their prereq tests.

Commit:
Add required params

A test(er) depending on another tester of a different type, now requires
that specific test to have those same default params added to the test
so that it is properly skipped.

Remove depreacted petsc min version from PetscJacobianTester

Commit:
tests which cannot run with --recover

Tests which cannot run with --recover options, due to prereq test
not being able to run with --recover.

milljm commented:
Closing in favor of #26329
